OPAL/tac/src/main/scala/org/opalj/tac/cg/android/AndroidEntryPointsFinder.scala

```diff
@@ -0,0 +1,52 @@
+/* BSD 2-Clause License - see OPAL/LICENSE for details. */
+package org.opalj
+package tac
+package cg
+package android
+
+import scala.collection.mutable.ArrayBuffer
+import scala.jdk.CollectionConverters.CollectionHasAsScala
+
+import org.opalj.br.Method
+import org.opalj.br.MethodDescriptor
+import org.opalj.br.ReferenceType
+import org.opalj.br.analyses.SomeProject
+import org.opalj.br.analyses.cg.EntryPointFinder
+import org.opalj.tac.fpcf.analyses.MethodDescription
+
+/**
+ * The AndroidEntryPointFinder considers specific methods of launcher Activity Clases as entry points.
+ * An activity is a launcher activity if it contains an intent filter with action "android.intent.action.MAIN"
+ * and category "android.intent.category.LAUNCHER". Requires Android Manifest to be loaded.
+ *
+ * @author Tom Nikisch
+ *         Julius Naeumann
+ */
+object AndroidEntryPointsFinder extends EntryPointFinder {
+    val configKey = "org.opalj.tac.cg.android.AndroidEntryPointsFinder.entryPoints"
+    override def collectEntryPoints(project: SomeProject): Iterable[Method] = {
+        val entryPointDescriptions = getConfiguredEntryPoints(project)
+        val manifest: AndroidManifest = project.get(AndroidManifestKey).get
+        // get launcher activities
+        val launchableActivities = manifest.components.collect { case a: Activity => a }.filter(_.isLauncherActivity)
```

Comment:
The collect here would not be necessary if components were stored separately on the manifest